avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/ArrayVisitor.java

```diff
@@ -35,8 +37,8 @@ public Schema builtAvroSchema() {
             throw new IllegalStateException("No element schema created for: "+_type);
         }
         Schema schema = Schema.createArray(_elementSchema);
-        if (_type.isArrayType()) {
-            schema.addProp(AVRO_SCHEMA_PROP_CLASS, _type.toCanonical());
+        if (!_type.hasRawClass(List.class)) {
+            schema.addProp(AVRO_SCHEMA_PROP_CLASS, AvroSchemaHelper.getTypeId(_type));
```

Comment:
Hmmh. Ok, but how about Scala's `List` equivalent? (it doesn't implement `List` but behaves similarly) Does `List` not need it, or is this actively harmful? My general worry is just that exact check for JDK type could be fragile, but I guess it's more due to my not understanding the need for this check.